waiter/src/waiter/scheduler.clj

```diff
@@ -665,6 +666,32 @@
                         service
                         (assoc instances :active-instances active-instances))))))))
 
+(defn start-new-services-handler
+  [token-metric-chan-mult start-new-service-fn leader?-fn fallback-state-atom]
+  (cid/with-correlation-id
+    "start-new-services-goroutine"
+    (let [correlation-id (cid/get-correlation-id)
+          process-token-event-ch-buffer (async/sliding-buffer 1000)
+          process-token-event-ch
+          (async/chan
+            process-token-event-ch-buffer
+            (comp
+              (filter (fn leader? [_]
+                        (leader?-fn)))
+              (filter (fn latest-service-does-not-exist?
+                        [{{:keys [service-id]} :latest-descriptor}]
```

Comment:
Seems like unnecessarily burdening the metrics service components to retrieve the latest descriptor only to be used here. This component should be able to compute its own latest service ID from the token.

service-id + services-mapping gives us token.
We have existing functions to check if a service is stale.
If the service is stale, token can be used to calculate new service ID.

Reply:
how do we compute the service-id of the current service handling requests for the token to update the local-usage-agent?
https://github.com/twosigma/waiter/pull/1526/files#diff-38b64ff3d02023614b695361fd800397fe9420a12941e9aaae8f45d509d639e5R155

Reply:
Via `compute-descriptor`
https://github.com/twosigma/waiter/blob/6d71edddb64e8a588cea31e167ae5bb7e3f2cd9b/waiter/src/waiter/descriptor.clj#L280-L299